js/pokemon-filter.js

```diff
@@ -0,0 +1,32 @@
+const nameInput = document.getElementById('filter-name'); 
+const defenseInput = document.getElementById('filter-defense');
+const attackInput = document.getElementById('filter-attack');
+const type1Input = document.getElementById('filter-type1'); 
+const abilityInput = document.getElementById('filter-ability'); 
+
+const pokemonsFilter = {
+    init(onFilter) {
+
+        nameInput.addEventListener('keyup', function() {
+            onFilter(nameInput.value); 
```

Comment:
```suggestion
            onFilter(nameInput.value, defenseInput.value, attackInput.value, type1Input.value, abilityInput.value);
```
Currently, the filter looks like it works as intended but with the right combination of keystrokes, it will actually ignore certain user input.